nucypher/acumen/perception.py

```diff
@@ -93,24 +101,28 @@ def archived(self) -> ArchivedFleetState:
                                   timestamp=self.timestamp,
                                   population=self.population)
 
-    def _remote_nodes_updated(self,
-                              nodes_to_add: Iterable['Ursula'],
-                              nodes_to_remove: Iterable[ChecksumAddress]
-                              ) -> bool:
+    def _calculate_diff(self,
+                        this_node_updated: bool,
+                        nodes_to_add: Iterable['Ursula'],
+                        nodes_to_remove: Iterable[ChecksumAddress]
+                        ) -> StateDiff:
 
+        nodes_updated = []
         for node in nodes_to_add:
             if node.checksum_address in nodes_to_remove:
                 continue
-            if node.checksum_address not in self._nodes:
-                return True
-            if bytes(self._nodes[node.checksum_address]) != bytes(node):
-                return True
+            if (    node.checksum_address not in self._nodes
```

Comment:
There are some extra whitespace characters here.

Reply:
I was trying to align the conditions following the spirit of our codestyle... Perhaps this would be better?
```
if     (node.checksum_address not in self._nodes
        or bytes(self._nodes[node.checksum_address]) != bytes(node)):
    nodes_updated.append(node.checksum_address)
```
In my own code I'd go with
```
if (node.checksum_address not in self._nodes
        or bytes(self._nodes[node.checksum_address]) != bytes(node)):
    nodes_updated.append(node.checksum_address)
```

Reply:
I tend to use more lines in this kind of situation, but ill chalk it up to style:

```python
unknown = node.checksum_address not in self._nodes
outdated = bytes(self._nodes[node.checksum_address]) != bytes(node)

if unknown or outdated:
    nodes_updated.append(node.checksum_address)

Reply:
This variant does look better, but `outdated` line will fail if `unknown` is True (and also there's a minor performance hit of eager `bytes()` calculation)

Reply:
Went for a half of this solution. At least the condition fits in one line now.